runtime-modules/content/src/migrations.rs

```diff
@@ -0,0 +1,219 @@
+                                    permissions_v1.insert(ContentModerationAction::HideChannel);
+                                }
+                                ContentModerationActionV0::ChangeChannelFeatureStatus(feature) => {
+                                    permissions_v1.insert(
+                                        ContentModerationAction::ChangeChannelFeatureStatus(
+                                            *feature,
+                                        ),
+                                    );
+                                }
+                                ContentModerationActionV0::DeleteVideoAssets(b) => {
+                                    permissions_v1
+                                        .insert(ContentModerationAction::DeleteVideoAssets(*b));
+                                }
+                                ContentModerationActionV0::DeleteNonVideoChannelAssets => {
+                                    permissions_v1.insert(
+                                        ContentModerationAction::DeleteNonVideoChannelAssets,
+                                    );
+                                }
+                                ContentModerationActionV0::UpdateChannelNftLimits => {
+                                    permissions_v1
+                                        .insert(ContentModerationAction::UpdateChannelNftLimits);
+                                }
+                                ContentModerationActionV0::DeleteVideo => {}
+                                ContentModerationActionV0::DeleteChannel => {}
+                            });
+                            permissions_by_level.insert(*level, permissions_v1);
+                        });
+                    let mut group_created_successfully = false;
+                    let mut group = CuratorGroupRecord::try_create::<T>(
+                        old_value.active,
+                        &permissions_by_level,
+                    )
+                    .map_or_else(
+                        |_err| Default::default(),
+                        |v| {
+                            group_created_successfully = true;
+                            v
+                        },
+                    );
+                    if !group_created_successfully {
+                        // The group will be dropped
+                        log!(warn, "Failed to create a group");
+                        return None;
+                    }
+                    old_value.curators.into_iter().for_each(|(k, v)| {
+                        let result = group.try_add_curator::<T>(k, &v);
+                        if result.is_err() {
+                            log!(warn, "Failed to add curator to group");
+                        }
+                    });
+                    Some(group)
+                },
+            );
+
+            current.put::<Pallet<T>>();
+
+            <T as frame_system::Config>::BlockWeights::get().max_block
```

Comment:
Why is this set to max? and is this the right notion of max? is the idea here to exclude all other processing in this block, so as to avoid having to come up with an accurate weight function to begin with?

Reply:
I was going through a lot of the migrations which we are including the nara update and saw more than one returning this max block weight. And since one of the migrations was doing it, it become almost irrelevant what was returned here. 

As far as I know the amount of block space allocated for on_runtime_upgrade is not constrained, but accurate weight computation would be important to know if we were planning a migration of a large map for example. This way we could plan ahead and decide if the total weight was larger than what validators could produce in required time.

Found this relevant discussion https://github.com/paritytech/substrate/issues/10064

Reply:
But to answer your question more clearly, yes by returning [max_block](https://github.com/paritytech/substrate/blob/033d4e86cc7eff0066cd376b9375f815761d653c/frame/system/src/limits.rs#L203) in this (or any other migration) will ensure that the block being executed (the first block with the new runtime active) will prevent any extrinsics from being included. So only migrations, on_initialize and on_finalize logic will execute.

The value is constructed in https://github.com/Joystream/joystream/blob/14b0fa39825209dc66264123f017fc51cd33c07f/runtime/src/lib.rs#L186